rest/route/stats.go

```diff
@@ -453,6 +453,19 @@ func (tsh *taskStatsHandler) Run(ctx context.Context) gimlet.Responder {
 	if err != nil {
 		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "Failed to retrieve the task stats"))
 	}
+	if len(taskStatsResult) == 0 {
+		// Lookup last sync date for project
+		statsStatus, err := stats.GetStatsStatus(tsh.filter.Project)
+		if err != nil {
+			return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "Failed to retrieve stats status"))
+		}
+		if time.Since(statsStatus.LastJobRun).Hours() > time.Duration.Hours(24) {
```

Comment:
It seems wrong that this should error if it hasn't been generated in the last 24 hours. What if the user is asking for task stats from a week ago? Those are generated and could be used without error. I think we could use ProcessedTasksUntil to just make sure that the time range the user is asking for is valid.

Reply:
yes that makes more sense